lib/maths/CGammaRateConjugate.cc

```diff
@@ -658,11 +657,15 @@ class CLogMarginalLikelihood : core::CNonCopyable {
 
             LOG_TRACE(<< "numberSamples = " << m_NumberSamples);
 
-            m_Constant = m_PriorShape * std::log(m_PriorRate) -
-                         boost::math::lgamma(m_PriorShape) +
+            m_Constant = m_PriorShape * std::log(m_PriorRate) - std::lgamma(m_PriorShape) +
                          logVarianceScaleSum - logGammaScaledLikelihoodShape -
-                         nResidual * boost::math::lgamma(m_LikelihoodShape) +
-                         boost::math::lgamma(m_ImpliedShape);
+                         nResidual * std::lgamma(m_LikelihoodShape) +
+                         std::lgamma(m_ImpliedShape);
```

Comment:
This would have the potential to be, for example, infinity - infinity := NaN, i.e. `- std::lgamma(m_PriorShape)` + ... + nResidual * std::lgamma(m_LikelihoodShape)`. Therefore, I think the following should also check for NaN and fail the calculation in this case. Also, nothing is left in try catch block that can throw, so this can be removed.

Reply:
> I think the following should also check for NaN and fail the calculation

`isfinite()` fails for inite **and** NaN

Reply:
... except, the return type is wrong. I think we should really be returning `maths_t::E_FpFailed` in this case.

Reply:
I guess the cleanest thing might be to keep the check as is and use `CMathsFuncs::fpStatus` for this.